src/main.js

```diff
@@ -285,23 +297,29 @@ function handleClose(streams, children, childrenInfo) {
         if (aliveChildren.length === 0) {
             exit(exitCodes);
         }
-
-        if (config.killOthers && !othersKilled) {
-            othersKilled = true
+        if (!othersKilled) {
+          othersKilled = true;
```

Comment:
1. concurrently is run with `--kill-others-on-fail` set;
2. First process exits with success. Sets `othersKilled` to `true`, but doesn't kill others because exit code is 0;
3. Second process crashes. `--kill-others-on-fail` does nothing, because the control flag `othersKilled` is `true`.